### PR TITLE
rpcdaemon: fix trace state diff with CREATE and SELFDESTRUCT

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: silkworm/rpc/core/evm_trace.cpp [hunk 1]
```diff
@@ -1140,6 +1140,41 @@ silkworm::ByteView StateAddresses::get_code(const evmc::address& address) const
...
+    if (it_nonce != nonces_.end()) {
+        return true;
+    }
+
+    auto it_code = codes_.find(address);
+    if (it_code != codes_.end()) {
+        return true;
+    }
+
+    return initial_ibs_.exists(address);
```
Comment (canepat) [inline silkworm/rpc/core/evm_trace.cpp:1175]:
```suggestion
    if (balances_.contains(address) || nonces_.contains(address) || codes_.contains(address)) {
        return true;
    }

    return initial_ibs_.exists(address);
```

File: silkworm/rpc/core/evm_trace.cpp [hunk 2]
```diff
@@ -1140,6 +1140,41 @@ silkworm::ByteView StateAddresses::get_code(const evmc::address& address) const
...
+    }
+
+    auto it_nonce = nonces_.find(address);
+    if (it_nonce != nonces_.end()) {
+        nonces_.erase(it_nonce);
+    }
+    auto it_codes = codes_.find(address);
+    if (it_codes != codes_.end()) {
+        codes_.erase(it_codes);
+    }
```
Comment (canepat) [inline silkworm/rpc/core/evm_trace.cpp:1156]:
```suggestion
    balances_.erase(address);
    nonces_.erase(address);
    codes_.erase(address);
```

**Review by canepat — CHANGES_REQUESTED** [review]

**Review by chfast — COMMENTED** [review]
Do you have any kind of regression tests for such changes?

**Review by canepat — APPROVED** [review]

**Comment by lupin012** [comment]
> Do you have any kind of regression tests for such changes?

Yes I have some new rpc-integration-tests cover this scenarios and I have many other rpc-integration-tests ("to try") to  avoid regression. 
These changes are related only tracer implementation (to be more aligned to erigon/rpcdaemon), the change on intra-block-state are just a getter.

**Pull request merged**